### PR TITLE
API-1524-validate-redirects-and-forwards 

### PR DESCRIPTION
**Pull request by crolarlibertyva**
Don't redirect, send out error

### CODE REVIEW

File: oauth-proxy/oauthHandlers/authorizeHandler.js [hunk 1]
```diff
@@ -16,12 +16,12 @@ const authorizeHandler = async (config, redirect_uri, logger, issuer, dynamo, dy
   try {
     const oktaApp = await oktaClient.getApplication(client_id);
     if (oktaApp.settings.oauthClient.redirect_uris.indexOf(client_redirect) === -1) {
-      const errorParams = new URLSearchParams({
-        error: 'invalid_client',
+      res.status(400).json({
```
Comment (beaugrantham-va) [inline oauth-proxy/oauthHandlers/authorizeHandler.js:19]: Looks good. Can we add a test case for this?

Comment (crolarlibertyva) [reply oauth-proxy/oauthHandlers/authorizeHandler.js:19]: Will do.

File: oauth-proxy/tests/oauthHandlers.test.js [hunk 2]
```diff
@@ -378,6 +378,18 @@ describe('authorizeHandler', () => {
     await authorizeHandler(config, redirect_uri, logger, issuer, dynamo, dynamoClient, oktaClient, req, res, next);
     expect(res.statusCode).toEqual(400);
   })
+
+  it('Bad redirect_uri', async () => {
+    req.query = {
+      state: "fake_state", 
+      client_id: "clientId123", 
+      client_redirect: "http://localhost:8080/oauth/redirect",
```
Comment (beaugrantham-va) [inline oauth-proxy/tests/oauthHandlers.test.js:386]: I don't recognize this parameter. It should probably be removed.

Otherwise it looks good.

Comment (crolarlibertyva) [reply oauth-proxy/tests/oauthHandlers.test.js:386]: Will do. I'll take care of happy case too.

**Pull request merged**